drivers/wifi/eswifi/eswifi_bus_spi.c

```diff
@@ -182,15 +182,34 @@ static int eswifi_spi_request(struct eswifi_dev *eswifi, char *cmd, size_t clen,
 
 static void eswifi_spi_read_msg(struct eswifi_dev *eswifi)
 {
+	const char startstr[] = "[SOMA]";
+	const char endstr[] = "[EOMA]";
 	char cmd[] = "MR\r";
+	size_t msg_len;
 	char *rsp;
-	int err;
+	int ret;
+
+	LOG_DBG("");
 
 	eswifi_lock(eswifi);
 
-	err = eswifi_at_cmd_rsp(eswifi, cmd, &rsp);
-	if (err < 0) {
-		LOG_ERR("Unable to read msg %d", err);
+	ret = eswifi_at_cmd_rsp(eswifi, cmd, &rsp);
+	if (ret < 0) {
+		LOG_ERR("Unable to read msg %d", ret);
+		eswifi_unlock(eswifi);
+		return;
+	}
+
+	if (strncmp(rsp, startstr, strlen(startstr))) {
```

Comment:
Instead of strlen(), it would be better to do `sizeof(startstr) - 1` as the latter is done at compile time.

Reply:
done